Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -107,253 +26,13 @@ variable [Semiring R] [∀ i, AddCommMonoid (M i)] [∀ i, AddCommMonoid (M₁ i
-      dite_comp_equiv_update (s.orderIsoOfFin hk).toEquiv]
-    simp
-
-/-- In the specific case of multilinear maps on spaces indexed by `Fin (n+1)`, where one can build
-an element of `∀ (i : Fin (n+1)), M i` using `cons`, one can express directly the additivity of a
-multilinear map along the first variable. -/
-theorem cons_add (f : MultilinearMap R M M₂) (m : ∀ i : Fin n, M i.succ) (x y : M 0) :
-    f (cons (x + y) m) = f (cons x m) + f (cons y m) := by
-  simp_rw [← update_cons_zero x m (x + y), f.map_update_add, update_cons_zero]
-
-/-- In the specific case of multilinear maps on spaces indexed by `Fin (n+1)`, where one can build
-an element of `∀ (i : Fin (n+1)), M i` using `cons`, one can express directly the multiplicativity
-of a multilinear map along the first variable. -/
-theorem cons_smul (f : MultilinearMap R M M₂) (m : ∀ i : Fin n, M i.succ) (c : R) (x : M 0) :
-    f (cons (c • x) m) = c • f (cons x m) := by
-  simp_rw [← update_cons_zero x m (c • x), f.map_update_smul, update_cons_zero]
-
-/-- In the specific case of multilinear maps on spaces indexed by `Fin (n+1)`, where one can build
-an element of `∀ (i : Fin (n+1)), M i` using `snoc`, one can express directly the additivity of a
-multilinear map along the first variable. -/
-theorem snoc_add (f : MultilinearMap R M M₂)
-    (m : ∀ i : Fin n, M (castSucc i)) (x y : M (last n)) :
-    f (snoc m (x + y)) = f (snoc m x) + f (snoc m y) := by
-  simp_rw [← update_snoc_last x m (x + y), f.map_update_add, update_snoc_last]
-
-/-- In the specific case of multilinear maps on spaces indexed by `Fin (n+1)`, where one can build
-an element of `∀ (i : Fin (n+1)), M i` using `cons`, one can express directly the multiplicativity
-of a multilinear map along the first variable. -/
-theorem snoc_smul (f : MultilinearMap R M M₂) (m : ∀ i : Fin n, M (castSucc i)) (c : R)
-    (x : M (last n)) : f (snoc m (c • x)) = c • f (snoc m x) := by
-  simp_rw [← update_snoc_last x m (c • x), f.map_update_smul, update_snoc_last]
-
 section
 
 variable {M₁' : ι → Type*} [∀ i, AddCommMonoid (M₁' i)] [∀ i, Module R (M₁' i)]
 variable {M₁'' : ι → Type*} [∀ i, AddCommMonoid (M₁'' i)] [∀ i, Module R (M₁'' i)]
 
-/-- If `g` is a multilinear map and `f` is a collection of linear maps,
-then `g (f₁ m₁, ..., fₙ mₙ)` is again a multilinear map, that we call
-`g.compLinearMap f`. -/
-def compLinearMap (g : MultilinearMap R M₁' M₂) (f : ∀ i, M₁ i →ₗ[R] M₁' i) :
-    MultilinearMap R M₁ M₂ where
-  toFun m := g fun i => f i (m i)
-  map_update_add' m i x y := by
-    have : ∀ j z, f j (update m i z j) = update (fun k => f k (m k)) i (f i z) j := fun j z =>
-      Function.apply_update (fun k => f k) _ _ _ _
-    simp [this]
-  map_update_smul' m i c x := by
-    have : ∀ j z, f j (update m i z j) = update (fun k => f k (m k)) i (f i z) j := fun j z =>
-      Function.apply_update (fun k => f k) _ _ _ _
-    simp [this]
-
 @[simp]
 theorem compLinearMap_apply (g : MultilinearMap R M₁' M₂) (f : ∀ i, M₁ i →ₗ[R] M₁' i)
```

Comment:
Same comment as from the other PRs; I don't think separating definitions from the lemma that unfolds them is helpful:

* it makes it harder to read the docs to work out what the function does
* it increases the risk that the unfolding lemma is stated in the wrong generality

Reply:
Agreed, especially when they look like `@[simps]` can generate them too. (This applies to quite a few results below too.)